### PR TITLE
Adding debug/exportz for MCS

### PR DESCRIPTION
**Pull request by nmittler**
Also doing some other misc cleanup.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -541,6 +541,16 @@ type ServiceDiscovery interface {
 
 	// NetworkManager returns a list of network gateways that can be used to access endpoints residing in this registry..
 	NetworkGateways() []*NetworkGateway
+
+	// ExportedServices returns a list of services exported by cluster.
+	ExportedServices() []ExportedService
```
Comment (howardjohn) [inline pilot/pkg/model/service.go:546]: It seems a bit heavy to add an interface to ServiceDiscovery, which is (and will be forever?) only implemented by a single controller and only for debug interface. Maybe we can just directly access the registry if it exists or something?

File: pilot/pkg/serviceregistry/kube/controller/serviceexportcache.go [hunk 2]
```diff
@@ -121,3 +143,14 @@ func (c disabledServiceExportCache) isExported(types.NamespacedName) bool {
 func (c disabledServiceExportCache) HasSynced() bool {
 	return true
 }
+
+func (c disabledServiceExportCache) ExportedServices() []model.ExportedService {
+	// MCS is disabled - show everything as exported.
+	return []model.ExportedService{
+		{
+			Cluster:   "*",
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/serviceexportcache.go:151]: We should document the `*` semantic somewhere?

Or maybe just return `nil` here, which would be different than `[]` which means no service disabled. That may be a bit subtle though..

Comment (nmittler) [reply pilot/pkg/serviceregistry/kube/controller/serviceexportcache.go]: Yeah I like `nil` better ... done.

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (nmittler) [reply pilot/pkg/model/service.go:546]: I don't disagree that it's a little heavy handed. However, it is just following the pattern established with gateways. I suppose it's possible that other types of clusters could provide some sort of MCS-like mechanism.

In general, I think I prefer this to having some sort of break-glass mechanism that directly accesses kube controllers. IDK ... WDYT?

@stevenctl any opinion?

Comment (stevenctl) [reply pilot/pkg/model/service.go:546]: I sort-of lean towards John's side here. I think if it's for debug purposes we can just do some casts from `ServiceDiscovery -> Aggregate` then call `GetRegistries` and cast to k8s controllers etc etc. 

If we were using this for actual generation logic I would probably lean the other way.

Comment (stevenctl) [reply pilot/pkg/model/service.go:546]: Or even implementing it in aggregate cluster and just leaving  it off the interface

**Comment by nmittler** [comment]
@howardjohn @stevenctl PTAL

**Review by stevenctl — APPROVED** [review]

**Pull request merged**